pkg/controller/workload/workload_cache.go

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package workload
+
+import (
+	"encoding/binary"
+	"sync"
+
+	workloadapi "kmesh.net/kmesh/api/v2/workloadapi"
+)
+
+var (
+	workloadCache = workloadStore{
+		byUid: cacheByUid{
+			cache: make(map[string]*workloadapi.Workload),
+		},
+		byAddr: cacheByAddr{
+			cache: make(map[NetworkAddress]*workloadapi.Workload),
+		},
+	}
+)
+
+type workloadStore struct {
+	byUid  cacheByUid
+	byAddr cacheByAddr
+}
+
+type NetworkAddress struct {
+	Network string
+	Address uint32
+}
+
+type cacheByUid struct {
+	cache map[string]*workloadapi.Workload
+	mutex sync.RWMutex
+}
+
+type cacheByAddr struct {
+	cache map[NetworkAddress]*workloadapi.Workload
```

Comment:
cacheByUid and cacheByAddr r/w should be atomic, that is to say we cannot use two locks

Reply:
cacheByUid and cacheByAddr sharing one lock at the same time? 
such as :
type workloadStore struct {
	byUid  cacheByUid
	byAddr cacheByAddr
        mutex sync.RWMutex
}

Reply:
fixed